File: 973-k-closest-points-to-origin/973.py
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        distance = []
        ans = []

        for i in range(len(points)):
            x,y = points[i]
            d = x*x+y*y
            distance.append([d, i])

        distance = sorted(distance)

        for i in range(k):
            ans.append(points[distance[i][1]])

        return ans

File: 973-k-closest-points-to-origin/973.cpp
class Solution {
public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        vector<pair<int, unsigned long>> vec;
        for (size_t i = 0; i < points.size(); ++i) {
            vector<int>& point = points[i];
            unsigned long distance = point[0] * point[0] + point[1] * point[1];
            vec.push_back({i, distance});
        }

        sort(vec.begin(), vec.end(), [](const auto& a, const auto& b) {
            return a.second < b.second;
        });

        vector<vector<int>> ans;
        for (int i = 0; i < k; ++i) {
            int index = vec[i].first;
            ans.push_back(points[index]);
        }

        return ans;
    }
};
